packages/x-data-grid/src/components/GridScrollArea.tsx

```diff
@@ -62,21 +65,52 @@ const GridScrollAreaRawRoot = styled('div', {
   },
 }));
 
-function GridScrollAreaRaw(props: ScrollAreaProps) {
-  const { scrollDirection } = props;
+const offsetSelector = createSelector(
+  gridDimensionsSelector,
+  (dimensions, direction: ScrollAreaProps['scrollDirection']) => {
+    if (direction === 'left') {
+      return dimensions.leftPinnedWidth;
+    }
+    if (direction === 'right') {
+      return dimensions.rightPinnedWidth + (dimensions.hasScrollX ? dimensions.scrollbarSize : 0);
+    }
+    return 0;
+  },
+);
+
+function GridScrollAreaWrapper(props: ScrollAreaProps) {
+  const apiRef = useGridApiContext();
+  const [dragging, setDragging] = React.useState<boolean>(false);
+
+  const handleColumnHeaderDragStart = useEventCallback(() => {
+    setDragging(true);
+  });
+
+  const handleColumnHeaderDragEnd = useEventCallback(() => {
+    setDragging(false);
+  });
+
+  useGridApiEventHandler(apiRef, 'columnHeaderDragStart', handleColumnHeaderDragStart);
+  useGridApiEventHandler(apiRef, 'columnHeaderDragEnd', handleColumnHeaderDragEnd);
```

Comment:
`useGridApiEventHandler` doesn't need memoized functions (with `useEventCallback`), it already replicates the same logic that `useEventCallback` implements.